### PR TITLE
In place pixels

### PR DESCRIPTION
**Pull request by dforegger**
Ongoing prototyping of in-place-pixel modification for performance gains. Got stoplight back to performant. Moved layout and n_pixels to be in the class so they don't need to be repassed and pre-processing (such as determining the indexes to work on) can be done.

Note: seeing slowdown on the opc put call.

TODO: We should write a test to make sure individual effects are performant

### CODE REVIEW

File: python/effects/stop_the_light.py [hunk 1]
```diff
@@ -26,7 +26,8 @@
 BLACK = (0, 0, 0)
 
 class StopTheLight(Effect):
-    def __init__(self, strip_bottom=0, strip_top=15):
+    def __init__(self, layout, n_pixels, strip_bottom=5, strip_top=20):
+        Effect.__init__(self, layout, n_pixels)
```
Comment (jobevers) [inline python/effects/stop_the_light.py]: Thank you for not using super! super in __init__ is frowned upon: https://fuhm.net/super-harmful/

Comment (jobevers) [inline python/effects/stop_the_light.py]: is n_pixels and layout redundant?

File: python/hoe/scene_manager.py [hunk 2]
```diff
@@ -203,11 +208,11 @@ def __str__(self):
         # TODO: What's the python way of doing this?
         return "EffectDefinition(%s)" % self.name
```
Comment (jobevers) [inline python/hoe/scene_manager.py]: "{}({})".format(self.__class__.__name__, self.name)

File: python/effects/stop_the_light.py (hunk 1 above)
Comment (dforegger) [inline python/effects/stop_the_light.py]: Just a really common operation avoiding extra len() functions. But yeah, we could just take layout and do len on it in Effect#__init__

**Pull request merged**